Add 3 sections

asankov opened the pull request:
- build
- Docker build
- Docker scan

There are some TODOs that will be resolved before merging.

I have not added any numbers to the sections, or links, because the order will probably change when we merge this with @SvetlomirBalevski 's work.

asankov committed:
3 sections

SvetlomirBalevski commented on 0x-docker-build/README.md:
```diff
@@ -0,0 +1,49 @@
+
+Paste the following `stage` snippet after the two stages from the previous section:
+
+
```
Docker pipeline plugin should be installed first (https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/Fix-Jenkins-Invalid-agent-type-Docker-specified-any-label-none-error).

We can leave it without explanation if we decide to reduce hand holding.

asankov replied:
if we merge https://github.com/SvetlomirBalevski/secure-pipeline-example/pull/1 the docker pipeline plugin will be installed alongside Jenkins

asankov committed:
Add info about creating Jenkins secrets

asankov committed:
Add section numbering

and links to next section

asankov committed:
Move build after SCA and SAST

move DAST as final step 8. After container image scan and "run" (that does not exist yet)

asankov committed:
Use `asankov/grype-opa:0.1` container

asankov committed:
Reformt `Build` section according to order

asankov committed:
Modify Section 0 and 1

SvetlomirBalevski committed:
Delete terraform.tfstate.backup

No need to push tf state file.
